libs/base/Data/Either.idr

```diff
@@ -31,6 +31,52 @@ isRight : Either a b -> Bool
 isRight (Left _)  = False
 isRight (Right _) = True
 
+--------------------------------------------------------------------------------
+-- `Either` used to a instance of interface `Foldable`.
```

Comment:
There is already an implementation of `Foldable (Either e)` in `Prelude.Types`.
There is no need to add all of these definitions again.
There is no need to add all the `%allow_overloads` you have inserted.

Reply:
As in [1](https://github.com/idris-lang/Idris2/pull/832#discussion_r539712524), I am wondering how to define the implementation of `null` for `(Either e)` and still don't know yet

Reply:
How does people actually use `null (x : Either a b)` as `isLeft`? Intuitively it can not be null for the name either :thinking:

Reply:
`null` for `t a` refers to whether or not `t a` contains an `a`.
With `Either e a`, `Left e` contains an `e` but not an `a`.

Reply:
Got that, thx u